src/get-virtualization-type.js

```diff
@@ -0,0 +1,39 @@
+let log = require('./log');
+let debug = log.debugCompat('aws-provisioner:lib:check-for-ami');
+
+/**
+ * Check for the existence of an AMI that's executable by
+ * the provided ec2 object in the given region
+ */
+module.exports = async function (ec2, ami) {
+  let request = {
+    /* Not sure why this isn't working
+    ExecutableUsers: [
+      'self',
+      'all',
+    ],*/
+    ImageIds: [ami],
+  };
+
+  let result;
+  try {
+    result = await ec2.describeImages(request).promise();
+    debug('loaded information about %s', ami);
+  } catch (err) {
+    debug('did not find %s', ami);
+    return false;
+  }
+
+  if (result.data.Images.length === 0) {
+    throw new Error('Image does not exist');
+  } else if (result.data.Images.length > 1) {
+    debug(result.data);
+    throw new Error('Image returned more than one result');
+  }
+
+  if (result.data.Images[0].ImageId === ami) {
+    return result.data.Images[0].VirtualizationType;
```

Comment:
This file is nearly identical to `ami-exists.js`.  I think we should consider merging them in that case.  In the places that I'm testing just for AMI existance, a return value of 
```javascript
return result.data.Images[0]
```
would pass the truthy or falsy questions I have of the answer to whether an ami exists, so we could probably merge the code into the same branch.  It would be good to rebase onto the `changes-22-8-2016` branch as that's going to be landing first thing in the Berlin.